subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -300,64 +336,95 @@ public FileCollection getAgentClasspath() {
      * Gets all properties in the format expected of the agent JVM argument.
      *
      * @return state of extension in a JVM argument
+     * @deprecated Use {@link #asJvmArgs()} instead.
      */
+    @Deprecated
```

Comment:
This deprecation needs some explanation.  The way to do this is to add a section in the upgrade guide: `upgrading_version_8.adoc` and create a deprecation warning message when this method is called.  Before returning add:

```
DeprecationLogger.deprecateAction("getAsJvmArg()")
                .willBeRemovedInGradle9()
                .withUpgradeGuideSection(8, "get_as_jvm_arg_deprecated")
                .nagUser()
```

and then write a short section called `get_as_jvm_arg_deprecated` in the guide explaining why and the replacement method to use.

Reply:
While looking for examples in the code base, I just found that `DeprecationLogger#deprecateMethod` also exists and seems more suitable in this case.